### PR TITLE
Default PHP exception handler will now fire

### PR DESCRIPTION
**Pull request by rjharrison**
This is a PR for [issue 523](https://github.com/bugsnag/bugsnag-php/issues/523) (And accidentally fixes [issue 475](https://github.com/bugsnag/bugsnag-php/issues/475))

In both cases, the root cause is that BugSnag **does** change default PHP behaviour with regard to exception handling and `registerWithPrevious` when no previous exception handler has been set.

It is more obvious in the CLI context, where the exit status from a script encountering an unhandled exception is incorrectly 0 ([issue 523](https://github.com/bugsnag/bugsnag-php/issues/523)). Without BugSnag enabled, the default PHP behaviour would be to generate a fatal error (`E_ERROR`) and the CLI script would exit with 255.

In a SAPI context, this is less obvious. Assuming `display_errors = off` and no content has been sent to the browser (ie output buffering is enabled), PHP default behaviour is to show a white screen of death with an HTTP status code of **500**. With BugSnag enabled via `registerWithPrevious`, the behaviour is to display a white screen of death with an HTTP status code of **200**. This is effectively the same thing as the CLI bug, but it's rarely noticeable unless something (a proxy or upstream server like Apache) interrogates the HTTP status code.

The additional effect that @thedotedge reported in [issue 475](https://github.com/bugsnag/bugsnag-php/issues/475) is that the default behaviour produces a fatal error that can be "trapped" (ish) within the shutdown function. When BugSnag is enabled via `registerWithPrevious` this error is never created and his switch statement fails (he makes a legitimate point that this is a bug).

In summary, if no previous exception handler exists, and the user invokes BugSnag via `registerWithPrevious`, then BugSnag should trigger the default exception handler in PHP. 

Note: a side effect to this is that BugSnag will now report on two events, the unhandled exception and the resulting fatal error. Perhaps the `E_ERROR` is something that could be filtered on the reporting side of things (the timestamp looks like it's the same for both events?)
Screenshot 2019-09-02 at 22 04 01

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Thanks for the PR. My only concern is that this may break a lot of existing code.

**Comment by rjharrison** [comment]
You know the product usage better than me, but at the point of an unhandled exception, without any exception handlers already registered, the PHP app is blowing up beyond recovery - it's already pretty broken. Will "the thing that called the PHP" break with this change? Maybe yea - but I can't imagine there are many people relying on this behaviour intentionally. I think it's more likely that if you need to react to a fatal error outside of PHP (ie Apache http redirection/serving static error pages/reacting via CLI) that you'd try and work around this bug (like forcing a non-zero exit status on the CLI). 

Good luck with it :-)

**Comment by abigail-bugsnag** [comment]
@rjharrison thanks for your comment. We agree with @GrahamCampbell that this will probably have a significant impact, however we also agree that the current behaviour doesn't seem correct.
We will discuss this and get back to you.

**Comment by imjoehaines** [comment]
Thanks for the PR, @rjharrison 

I've just released [v3.23.1](https://github.com/bugsnag/bugsnag-php/releases/tag/v3.23.1) which should resolve this issue, as well as some other issues with the error/exception handler
